docs: Update quickstart page to structure things a little more for the novices

franciscojavierarceo opened the pull request:
# What does this PR do?
Another doc enhancement for https://github.com/meta-llama/llama-stack/issues/1818

Summary of changes:
- `docs/source/distributions/configuration.md`
   - Updated dropdown title to include a more user-friendly description.

- `docs/_static/css/my_theme.css`
   - Added styling for `<h3>` elements to set a normal font weight.

- `docs/source/distributions/starting_llama_stack_server.md`
   - Changed section headers from bold text to proper markdown headers (e.g., `##`).
   - Improved descriptions for starting Llama Stack server using different methods (library, container, conda, Kubernetes).
   - Enhanced clarity and structure by converting instructions into markdown headers and improved formatting.

- `docs/source/getting_started/index.md`
   - Major restructuring of the "Quick Start" guide:
      - Added new introductory section for Llama Stack and its capabilities.
      - Reorganized steps into clearer subsections with proper markdown headers.
      - Replaced dropdowns with tabbed content for OS-specific instructions.
      - Added detailed steps for setting up and running the Llama Stack server and client.
      - Introduced new sections for running basic inference and building agents.
      - Enhanced readability and visual structure with emojis, admonitions, and examples.

- `docs/source/providers/index.md`
   - Updated the list of LLM inference providers to include "Ollama."
   - Expanded the list of vector databases to include "SQLite-Vec."

Let me know if you need further details!

## Test Plan
Renders locally, included screenshot.

# Documentation

For https://github.com/meta-llama/llama-stack/issues/1818

<img width="1332" alt="Screenshot 2025-04-09 at 11 07 12 AM" src="https://github.com/user-attachments/assets/c106efb9-076c-4059-a4e0-a30fa738585b" />

raghotham reviewed (changes requested):
the documentation changes look good. but this PR is changing many more files - is that intentional?

franciscojavierarceo commented:
> the documentation changes look good. but this PR is changing many more files - is that intentional?

No, I'll fix rebase this, something went wrong when I made a new branch. Will be cleaned up later this evening. Sorry about that.

franciscojavierarceo commented:
@raghotham I incorporated the latest changes and fixed the conflicts. Still gives the same structure (i.e., making a install and setup at the beginning).

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,87 +1,92 @@
 
-Now you can use the llama stack client to run inference and build agents!
+### ii. Using the Llama Stack Client
+Now you can use the llama stack client to run inference and build agents! You can reuse the server setup or use the
```
can we make installing a client package or reusing server setup be tab-items?

raghotham commented on docs/source/getting_started/index.md:
````diff
@@ -417,23 +415,24 @@ ragagent = Agent(
+uv run python rag_agent.py
 ```
-:::{dropdown} `Sample output`
+Which will output:
````
This output is too long - it was dropdown because i didnt want the doc length to grow. Maybe we can have 3 tabs for each example - script, run, output? That way, we wont have to have a really long docs page.

franciscojavierarceo replied:
This output was only 10 lines, so I thought just displaying the output was fine. I'll switch it back to the dropdown 👍

I think the 3 tabs for script, run, output actually may be less intuitive for readers. As often readers will expect to move through examples sequentially (i.e., vertically) rather than horizontally.

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,87 +1,92 @@
-### 2. Run Llama Stack locally
-
-We use `uv` to setup a virtual environment and install the Llama Stack package.
+### ii. Install `uv` to Manage your Python packages
```
how much prominence should we give to setting up python. Should we just point to the appropriate external docs instead of showing them here. Also, would it be good to use tabs to show different envs like venv/conda/docker?

franciscojavierarceo replied:
It's just a tradeoff. By linking it, we're implicitly saying we'll maintain the docs in favor of an easier onboarding for developers and I think that's the right move given the current stage of the project.

I'll add the conda/docker configurations as well.

franciscojavierarceo commented:
@raghotham updated 👍 

I'll open the other PR after this is closed to show the smaller demo for getting started.

Bobbins228 commented on docs/source/getting_started/index.md:
````diff
@@ -1,87 +1,174 @@
 
+```{admonition} Note
+:class: tip
+These instructions outlined are for a
````
An incomplete sentence here

Bobbins228 commented on docs/source/getting_started/index.md:
```diff
@@ -1,87 +1,174 @@
+`podman`. If you are using `podman` older than `4.7.0`, please also replace `host.docker.internal` in the `OLLAMA_URL`
+with `host.containers.internal`.
+
+As another example, to start the container with Podman, you can do the same but replace `docker` at the start of the command with `podman`. If you are using `podman` older than `4.7.0`, please also replace `host.docker.internal` in the `OLLAMA_URL` with `host.containers.internal`.
```
Duplication of the above note

raghotham commented on docs/source/getting_started/index.md:
````diff
@@ -1,87 +1,175 @@
+
+Then download a Llama model with Ollama
 ```bash
 ollama pull llama3.2:3b-instruct-fp16
````
change model to llama3.2:3b - its smaller and faster

franciscojavierarceo replied:
done

raghotham commented on docs/_static/css/my_theme.css:
```diff
@@ -16,3 +16,7 @@
     display: none;
 }
+
+h3 {
```
Should it be h2 instead of h3?

raghotham commented on docs/source/getting_started/detailed_tutorial.md:
```diff
@@ -0,0 +1,527 @@
+Llama Stack config for Ollama.
+::::{tab-set}
+
+:::{tab-item} Using Python
```
should add conda as well

raghotham commented on docs/source/getting_started/detailed_tutorial.md:
````diff
@@ -0,0 +1,527 @@
+
+First lets setup some environment variables and create a local directory to mount into the container’s file system.
+```bash
+export INFERENCE_MODEL="meta-llama/Llama-3.2-3B-Instruct"
````
inference model can be the same as the one used above - we dont need to introduce new names to the same models.

raghotham commented on docs/source/getting_started/detailed_tutorial.md:
```diff
@@ -0,0 +1,527 @@
+You can reuse the server setup or use the [Llama Stack Client](https://github.com/meta-llama/llama-stack-client-python/).
+Note that the client package is already included in the `llama-stack` package.
+
+### ii. Using the Llama Stack Client
```
there is no ### i.

raghotham commented on docs/source/getting_started/detailed_tutorial.md:
````diff
@@ -0,0 +1,527 @@
+
+```
+
+## Step 4: Run Basic Inference
````
the rest of the "steps" are just different examples.

raghotham commented on docs/source/getting_started/detailed_tutorial.md:
````diff
@@ -0,0 +1,527 @@
+You can test basic Llama inference completion using the CLI.
+
+```bash
+llama-stack-client inference chat-completion --message "tell me a joke"
````
can move this to the CLI Section above - we are not going to have any other example with client cli

raghotham commented on docs/source/getting_started/detailed_tutorial.md:
````diff
@@ -0,0 +1,527 @@
+```
+Alternatively, you can run inference using the Llama Stack client SDK.
+
+### i. Create the Script
````
seems too heavy weight to have these be section headers. Or, maybe the css works?

raghotham commented on docs/source/getting_started/index.md:
````diff
@@ -1,455 +1,106 @@
-Setup venv
-```bash
-uv venv --python 3.10
+uv pip install llama-stack aiosqlite faiss-cpu ollama \
````
why are so many packages needed? why not just pip install llama-stack?

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,455 +1,106 @@
+uv pip install llama-stack aiosqlite faiss-cpu ollama \
+openai datasets opentelemetry-exporter-otlp-proto-http mcp autoevals
 source .venv/bin/activate
+export INFERENCE_MODEL="llama3.2:3b"
```
why have the env var here?

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,455 +1,106 @@
-from llama_stack_client import LlamaStackClient
-from llama_stack_client import Agent, AgentEventLogger
+from termcolor import cprint
+from llama_stack.distribution.library_client import LlamaStackAsLibraryClient
```
we dont want library as client - let's use server please.

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,455 +1,106 @@
-]
+vector_db = "faiss"
+vector_db_id = "test-vector-db"
+model_id = "llama3.2:3b-instruct-fp16"
```
wrong model id - it should be llama3.2:3b

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,455 +1,106 @@
+client.vector_dbs.register(
+    provider_id=vector_db,
+    vector_db_id=vector_db_id,
+    embedding_model="all-MiniLM-L6-v2",
```
embedding model showed up here - can we retrieve it from the distro server?

raghotham commented on docs/source/getting_started/index.md:
```diff
@@ -1,455 +1,106 @@
-PyTorch Tune provides a recipe for LoRA (Low-Rank Adaptation) finetuning, which is a technique to adapt pre-trained models to new tasks. The recipe uses the `lora_finetune_distributed` command.
-...
-Overall, DORA is a powerful reinforcement learning algorithm that can learn complex tasks from human demonstrations. However, it requires careful consideration of the challenges and limitations to achieve optimal results.
+--------------------------------------------------
```
we are missing the G of RAG! :)

raghotham commented:
will make minor fixes after merging this PR
